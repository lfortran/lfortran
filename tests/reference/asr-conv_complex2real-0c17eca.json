{
    "basename": "asr-conv_complex2real-0c17eca",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/conv_complex2real.f90",
    "infile_hash": "45bd43af8695005b07db0e167445c8230cda8e3201e8ce644bf187d2",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-conv_complex2real-0c17eca.stdout",
    "stdout_hash": "b974a1d0f8de499ca9b075d6413e4ab661fb1afe940ea01a08c101ab",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}