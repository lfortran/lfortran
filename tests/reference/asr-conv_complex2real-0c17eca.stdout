(TranslationUnit
    (SymbolTable
        1
        {
            conv_complex2real:
                (Program
                    (SymbolTable
                        2
                        {
                            x:
                                (Variable
                                    2
                                    x
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            z:
                                (Variable
                                    2
                                    z
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Complex 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    conv_complex2real
                    []
                    [(Assignment
                        (Var 2 z)
                        (ComplexConstructor
                            (RealConstant
                                3.000000
                                (Real 4)
                            )
                            (RealConstant
                                4.000000
                                (Real 4)
                            )
                            (Complex 4)
                            (ComplexConstant
                                3.000000
                                4.000000
                                (Complex 4)
                            )
                        )
                        ()
                    )
                    (Assignment
                        (Var 2 x)
                        (Cast
                            (Cast
                                (Var 2 z)
                                ComplexToReal
                                (Real 8)
                                ()
                            )
                            RealToReal
                            (Real 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (RealCompare
                            (IntrinsicElementalFunction
                                Abs
                                [(RealBinOp
                                    (Var 2 x)
                                    Sub
                                    (RealConstant
                                        3.000000
                                        (Real 4)
                                    )
                                    (Real 4)
                                    ()
                                )]
                                0
                                (Real 4)
                                ()
                            )
                            Gt
                            (RealConstant
                                0.000010
                                (Real 4)
                            )
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                )
        })
    []
)
